### PR TITLE
Bluetooth: Host: Discard advertising data if not explicit scanning

### PR DESCRIPTION
**Pull request by Thalley**
If the application is not explicitly scanning, then there is not really any need to parse advertising reports nor send them to the application.

### CODE REVIEW

**Comment by Thalley** [comment]
This is a change in behavior and ideally would be in the release notes for 4.0, but they don't exist yet

**Comment by Thalley** [comment]
Marking as draft until tests passing

**Review by jhedberg — DISMISSED** [review]

**Comment by Thalley** [comment]
Moved the check to the loop in case that an application disables scan in the scan recv callback. 

@jhedberg or others: What is the purpose of https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/scan.c#L509-L511 ? This change won't call that in the case that we are doing an implicit scan.

**Review by kruithofa — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

File: tests/bluetooth/host_long_adv_recv/src/main.c [hunk 1]
```diff
@@ -201,6 +205,7 @@ static void le_read_supp_states(struct net_buf *buf, struct net_buf **evt, uint8
 	(void)memset(&rp->le_states, 0xFF, sizeof(rp->le_states));
 }
 
+
```
Comment (jhedberg) [inline tests/bluetooth/host_long_adv_recv/src/main.c:208]: This seems like an unrelated (and unnecessary) change?

**Review by jori-nordic — APPROVED** [review]
